src/main/resources/EDIFACT-SupplyChain-D03B/EDIFACT-SupplyChain-Messages-D.03B.xsd

```diff
@@ -87,99 +87,82 @@
               </xsd:element>
-              <xsd:element ref="srv:KEYMAN">
+              <xsd:element ref="srv:KEYMAN" dfdl:choiceBranchKey="keyman">
                 <xsd:annotation>
                   <xsd:appinfo source="http://www.ogf.org/dfdl/">
-                    <dfdl:discriminator>
-                      {fn:contains(../UNH/S009/E0065,'KEYMAN')}
-                    </dfdl:discriminator>
                   </xsd:appinfo>
                 </xsd:annotation>
               </xsd:element>
-              <xsd:element ref="D03B:ORDCHG">
+              <xsd:element ref="D03B:ORDCHG" dfdl:choiceBranchKey="ordchg">
                 <xsd:annotation>
                   <xsd:appinfo source="http://www.ogf.org/dfdl/">
-                    <dfdl:discriminator>
-                      {fn:contains(../UNH/S009/E0065,'ORDCHG')}
-                    </dfdl:discriminator>
                   </xsd:appinfo>
                 </xsd:annotation>
               </xsd:element>
-              <xsd:element ref="D03B:ORDERS">
+              <xsd:element ref="D03B:ORDERS" dfdl:choiceBranchKey="orders">
                 <xsd:annotation>
                   <xsd:appinfo source="http://www.ogf.org/dfdl/">
-                    <dfdl:discriminator>
-                      {fn:contains(../UNH/S009/E0065,'ORDERS')}
-                    </dfdl:discriminator>
                   </xsd:appinfo>
                 </xsd:annotation>
               </xsd:element>
-              <xsd:element ref="D03B:ORDRSP">
+              <xsd:element ref="D03B:ORDRSP" dfdl:choiceBranchKey="ordrsp">
                 <xsd:annotation>
                   <xsd:appinfo source="http://www.ogf.org/dfdl/">
-                    <dfdl:discriminator>
-                      {fn:contains(../UNH/S009/E0065,'ORDRSP')}
-                    </dfdl:discriminator>
                   </xsd:appinfo>
                 </xsd:annotation>
               </xsd:element>
-              <xsd:element ref="D03B:RECADV">
+              <xsd:element ref="D03B:RECADV" dfdl:choiceBranchKey="recadv">
                 <xsd:annotation>
                   <xsd:appinfo source="http://www.ogf.org/dfdl/">
-                    <dfdl:discriminator>
-                      {fn:contains(../UNH/S009/E0065,'RECADV')}
-                    </dfdl:discriminator>
                   </xsd:appinfo>
                 </xsd:annotation>
               </xsd:element>
-              <xsd:element ref="D03B:REMADV">
+              <xsd:element ref="D03B:REMADV" dfdl:choiceBranchKey="remadv">
                 <xsd:annotation>
                   <xsd:appinfo source="http://www.ogf.org/dfdl/">
-                    <dfdl:discriminator>
-                      {fn:contains(../UNH/S009/E0065,'REMADV')}
-                    </dfdl:discriminator>
                   </xsd:appinfo>
                 </xsd:annotation>
               </xsd:element>
       
-              <xsd:sequence>
+              <xsd:sequence dfdl:choiceBranchKey="badmsg">
```

Comment:
This is the "default" case if none of the other choiceBranchKeys match. The best way to handle this is not to make the choiceDispatchKey compute an explicit tag for this, but to use two nested choices:

```
<choice> <!-- a choice between recognized messages, and badmsg -->
    <choice dfdl:choiceDispatchKey="......">
         .... branches for real messages here....
    </choice>
   <element name="badmsg" ...../><!-- if none of the above inner choice are matched --> 
</choice>
```